### PR TITLE
AX-476: [Sell Dialog] Implement GraphQl with BloC for APTSellInfo

### PR DESCRIPTION
**Pull request by bekchanovj**
# Description
Set up a BloC for the Sell Dialog. Added a model for SellInfo and BuyInfo. 

Fixes Jira Ticket # 
476
https://athletex.atlassian.net/browse/AX-476?atlOrigin=eyJpIjoiNGFjNGZjZTIzYWQzNDMzZGI2MTliZmM3NzJkM2IzMTAiLCJwIjoiaiJ9

## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Tested in the debug mode using Google Chrome browser. 

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have assigned 2 reviewers to check my work
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: lib/service/BlockchainModels/AptSellInfo.dart [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class AptSellInfo {
```
Comment (ketonkss4) [inline lib/service/BlockchainModels/AptSellInfo.dart:1]: This is nearly the same thing as AptBuyInfo, lets consolidate the models to AptTransactionInfo or something along those lines

File: lib/dialogs/sell/usecases/GetAPTSellInfoUseCase.dart [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import 'package:ax_dapp/dialogs/buy/usecases/GetSwapInfoUseCase.dart';
+import 'package:ax_dapp/service/BlockchainModels/AptSellInfo.dart';
+import 'package:ax_dapp/service/Controller/Swap/AXT.dart';
+import 'package:fpdart/fpdart.dart';
+
+const String _no_sell_info_error_msg = "No sell info found";
+
+class GetAPTSellInfoUseCase {
```
Comment (ketonkss4) [inline lib/dialogs/sell/usecases/GetAPTSellInfoUseCase.dart:8]: Looks very similar to GetBuyInfoUseCase... Perhaps we can consolidate the two? Buy is a swap between the targetApt and Ax so maybe the GetBuyInfoUseCase should just call this with a hardcoded tokenFrom value

File: lib/dialogs/sell/bloc/SellDialogBloc.dart [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+      final response =
+          await repo.fetchAptSellInfo(aptAddress: state.tokenAddress!);
+      final isSuccess = response.isLeft();
+
+      if (isSuccess) {
+        print(" New Apt Input: Success");
+        if (swapController.amount1.value != aptInputAmount) {
+          swapController.updateFromAmount(aptInputAmount);
+        }
+        final buyInfo = response.getLeft().toNullable()!.aptSellInfo;
```
Comment (ketonkss4) [inline lib/dialogs/sell/bloc/SellDialogBloc.dart:109]: The copy is showing here with your variable name choice

**Review by ketonkss4 — CHANGES_REQUESTED** [review]

**Review by pearlson16 — DISMISSED** [review]

File: lib/dialogs/sell/bloc/SellDialogBloc.dart [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    final minimumReceiveAmt = receiveAmount * (1 - _slippage_tolerance);
+    return TransactionInfo(
+        minimumReceiveAmt, priceImpact, receiveAmount, totalFees, price);
+  }
+
+  void _mapMaxSellTapEventToState(
+      MaxSellTap event, Emitter<SellDialogState> emit) async {
+    emit(state.copy(status: Status.loading));
+    try {
+      final maxInput = await wallet.getTotalAxBalance();
```
Comment (mretana1999) [inline lib/dialogs/sell/bloc/SellDialogBloc.dart:92]: should this fetch the current APT balance?

**Review by mretana1999 — APPROVED** [review]

**Review by pearlson16 — APPROVED** [review]

**Pull request merged**